pkgs/by-name/wr/wrangler/package.nix

```diff
@@ -0,0 +1,109 @@
+{
+  lib,
+  pkgs,
+  stdenv,
+  fetchFromGitHub,
+  pnpm_9,
+  nodejs,
+  npmHooks,
+  writeScriptBin
+}:
+let
+  srcHash = "sha256-/4iIkvSn85fkRggmIha2kRlW0MEwvzy0ZAmIb8+LpZQ=";
+  pnpmDepsHash = "sha256-aTTaiGXm1WYwmy+ljUC9yO3qtvN20SA+24T83dWYrI0=";
+in
+  stdenv.mkDerivation (finalAttrs: {
+  pname = "wrangler";
+  version = "3.62.0";
+
+  src = fetchFromGitHub {
+    owner = "cloudflare";
+    repo = "workers-sdk";
+    rev = "wrangler@${finalAttrs.version}";
+    hash = "${srcHash}";
+  };
+
+  buildInputs = [
+    pkgs.llvmPackages.libcxx
+    pkgs.llvmPackages.libunwind
+    pkgs.musl
+    pkgs.xorg.libX11
+  ] ++ lib.optional stdenv.isLinux pkgs.autoPatchelfHook;
+
+  nativeBuildInputs = [
+    nodejs
+    pnpm_9.configHook
+    pkgs.autoPatchelfHook
+  ];
+
+  pnpmDeps = pnpm_9.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "${pnpmDepsHash}";
+  };
+
+  preBuild = ''
+    addAutoPatchelfSearchPath lib/node_modules/.pnpm/@cloudflare+workerd-linux-64@1.20240620.1/node_modules/@cloudflare/workerd-linux-64/bin/
+    '';
+
+  # @cloudflare/vitest-pool-workers wanted to run a server as part of the build process
+  # so I simply removed it
+  postBuild = ''
+    rm -fr packages/vitest-pool-workers
+    NODE_ENV="production" pnpm --filter miniflare run build
+    NODE_ENV="production" pnpm --filter wrangler run build
+    # pnpm --offline deploy --frozen-lockfile  --ignore-script  --filter=bash-language-server server-deploy
+    '';
+
+  # this was taken from a previous script which was generated somehow
+  wranglerScript = pkgs.writeText "wrangler" ''
+        #!${pkgs.bash}/bin/sh
+        basedir=$(dirname "$(echo "$0" | sed -e 's,\\,/,g')")
+
+        case `uname` in
+            *CYGWIN*) basedir=`cygpath -w "$basedir"`;;
+        esac
+
+        if [ -z "$NODE_PATH" ]; then
+            export NODE_PATH="WRANGLER_PATH/lib/node_modules:WRANGLER_PATH/lib/packages/wrangler/node_modules"
+        else
+            export NODE_PATH="WRANGLER_PATH/lib/node_modules:WRANGLER_PATH/lib/packages/wrangler/node_modules:$NODE_PATH"
+        fi
+        if [ -x "$basedir/node" ]; then
+            exec "$basedir/node" "WRANGLER_PATH/lib/packages/wrangler/bin/wrangler.js" "$@"
+        else
+            exec node "WRANGLER_PATH/lib/packages/wrangler/bin/wrangler.js" "$@"
```

Comment:
In general, it may very well be that this script can be generated with `makeWrapper`, see an example at the expression of `bash-language-server`'s `installPhase`.